In [1]:
!pip install -r requirements.txt

In [2]:
import numpy as np

# Sistema Masa Resorte

## Problema

Considera un sistema masa-resorte en un plano horizontal, en el cual un resorte está unido a un punto fijo 
y del otro extremo a una masa de 0.5 kg, se inicia el movimiento con el resorte estirado a 10 cm, por último, 
sabemos que la constante del resorte es de 50 N/m.
Determina el periodo del sistema.

Obtén:
* Posición
* Velocidad 
* Aceleración 

en un tiempo de un segundo.

In [3]:
m = 0.5 # masa en kg
k = 50 # constante del resorte en N/m
A = 0.1 # amplitud en m
t = 1 # tiempo en s para calcular la posición

# calculo de la frecuencia angular
omega = np.sqrt(k/m) # frecuencia angular en rad/s
print("Frecuencia angular:", f"{omega:.3f}", "rad/s")

# calculo del periodo
T = round(2 * np.pi * np.sqrt(m/k), 3) # periodo en s
print("Periodo: ", f"{T:.3f}", "s")

# calculo de la frecuencia
f = round(1/T, 3) # frecuencia en Hz
print("Frecuencia:", f, "Hz")

# calculo de la posición
x_t = round(A * np.cos(omega * t), 3)
print("Posición x(t):", x_t, "m")

# calculo de la velocidad en cm/s
v_t = round(-A * omega * np.sin(omega * t) * 100, 3)
print("Velocidad v(t):", v_t, "cm/s")

# calculo de la aceleración
a_t = round(-A * (omega ** 2) * np.cos(omega * t), 3)
print("Aceleración a(t): ", a_t, "m/s^2")


Frecuencia angular: 10.000 rad/s
Periodo:  0.628 s
Frecuencia: 1.592 Hz
Posición x(t): -0.084 m
Velocidad v(t): 54.402 cm/s
Aceleración a(t):  8.391 m/s^2


# Datos

$m = 0.5 kg$

$k = 50 N/m$

$A = 10cm = 0.1 m$

$t = 1 s$

$\phi = 0$

$\omega = \sqrt{\frac{k}{m}} = \sqrt{\frac{50}{0.5}} = 10 rad/s$

# Solución

## Operaciones en base al código previo

Calculo del periodo: 

$T = 2\pi\sqrt{\frac{m}{k}} = 2\pi\sqrt{\frac{0.5}{50}} = 0.628 s$


Calculo de la frecuencia: 

$f = \frac{1}{T} = \frac{1}{0.628} = 1.592 Hz$


Calculo de la posición: 

$X(t) = A\cos(\omega t + \phi) = 0.1\cos(10t + \phi) = 0.1\cos(10t) = 0.1\cos(10) = 0.1(-0.839) = -0.084 m$


Calculo de la velocidad: 

$v(t) = -A\omega\sin(\omega t + \phi) = -0.1(10)\sin(10t + \phi) = -0.1(10)\sin(10t) = -0.1(10)\sin(10) = -0.1(0.544) = -0.05402 m/s = 54.402 cm/s$


Calculo de la aceleracion: 

$a(t) = -A\omega^2\cos(\omega t + \phi) = -0.1(10)^2\cos(10t + \phi) = -0.1(10)^2\cos(10t) = -0.1(10)^2\cos(10) = -0.1(10)^2(-0.839) = -8.391 m/s^2$


-----------------

# Pendulo Simple

## Problema

Determina el periodo de un péndulo simple situado en el planeta Tierra que tiene una 
longitud de cuerda de 1.2 metros y después demuéstralo con el simulador de VASCAK 
https://www.vascak.cz/data/android/physicsatschool/template.php?s=kv_kyvadlo&l=es

In [4]:
L = 1.2 # longitud en m
g = 9.81 # gravedad en m/s^2

# calculo del periodo
T = round(2 * np.pi * np.sqrt(L/g), 3) # periodo en s
print("Periodo:", T, "s")

Periodo: 2.198 s


# Datos

$L = 1.2 m$

$g = 9.81 m/s^2$

# Solución

## Operaciones en base al código previo

Calculo del periodo: 

$T = 2\pi\sqrt{\frac{L}{g}} = 2\pi\sqrt{\frac{1.2}{9.81}} = 2.198 s$

Simulación:
![Pendulo](simulacion-pendulo-simple.png)

----

# Principio de Superposición de Ondas

## Problema

Dado el principio de superposición de onda resuelve el siguiente problema aplicado:

Se requiere conocer la sumatoria analítica de dos ondas mecánicas, las cuales están en el eje horizontal y con dirección hacia el este. 
Sabemos que las ondas mecánicas tienen la siguiente descripción matemática:

$y_1 = 3\sin(2x - 8t), y_2 = 3sin(-3x + 4t - \pi)$

# Solución

## Sumatoria analítica

$y_1 = 3\sin(2x - 8t), y_2 = 3sin(-3x + 4t - \pi)$

$y_{total}(x,t) = y_1(x,t) + y_2(x,t)$

$\therefore y_{total}(x,t) = 3\sin(2x - 8t) + 3sin(-3x + 4t - \pi)$

-----

# Efecto Doppler

Una ambulancia viaja al este por una carretera con velocidad 33 m/s; su sirena emite sonido con una frecuencia de 400 Hz. ¿Qué frecuencia escucha una persona en un auto que viaja al oeste con velocidad 24 m/s? De ser necesario guíate con el simulador de VASCAK y agrega imágenes donde representen la situación del actual problema https://www.vascak.cz/data/android/physicsatschool/template.php?f=kv_doppler&l=es
Cuando el auto se acerca a la ambulancia.
Cuando el auto se aleja de la ambulancia.

In [5]:
frecuencia_fuente = 400 # frecuencia en Hz
velocidad_sonido = 343 # velocidad del sonido en m/s
velocidad_ambulancia = 33 # velocidad de la ambulancia en m/s
velocidad_auto = 24 # velocidad del auto en m/s

# calculo de la frecuencia percibida por el observador
def frecuencia_doppler(frecuencia_fuente, velocidad_sonido, velocidad_observador, velocidad_fuente, acercando=True):
    if acercando:
        return frecuencia_fuente * ((velocidad_sonido + velocidad_observador) / (velocidad_sonido - velocidad_fuente))
    else:
        return frecuencia_fuente * ((velocidad_sonido - velocidad_observador) / (velocidad_sonido + velocidad_fuente))
    

# calculo de la frecuencia acercandose
frecuencia_acercandose = frecuencia_doppler(frecuencia_fuente, velocidad_sonido, velocidad_ambulancia, velocidad_auto, acercando=True)
print("Frecuencia acercandose:", round(frecuencia_acercandose, 3), "Hz")

# calculo de la frecuencia alejandose
frecuencia_alejandose = frecuencia_doppler(frecuencia_fuente, velocidad_sonido, velocidad_ambulancia, velocidad_auto, acercando=False)
print("Frecuencia alejandose:", round(frecuencia_alejandose, 3), "Hz")

Frecuencia acercandose: 471.473 Hz
Frecuencia alejandose: 337.875 Hz


# Datos

$f_{ambulancia} = 400 Hz$

$v_{sonido} = 343 m/s$

$v_{ambulancia} = 33 m/s$

$v_{auto} = 24 m/s$

# Solución

## Operaciones en base al código previo

Calculo de la frecuencia:

$f' = \frac{v_{sonido} \pm v_{auto}}{v_{sonido} \pm v_{ambulancia}} * f_{ambulancia}$

Acercandose:

$f' = \frac{343 + 24}{343 - 33} * 400 = 471.473 Hz$

Alejandose:

$f' = \frac{343 - 24}{343 + 33} * 400 = 337.875 Hz$

## Simulación

![Doppler Acercandose](efecto-doppler-acercandose.png)
![Doppler Alejandose](efecto-doppler-alejandose.png)

----

# Principio de Pascal

## Problema

En un taller mecánico cuentan con una prensa hidráulica, la cual tiene dos extremos, 
un pedal cuadrado de 10 cm de lado y el otro con un área de 3 m2, si en el pedal se 
para una persona de 90 kg, determina la fuerza que produce del otro lado.

Si del extremo del pedal se encuentra una persona de 75 kg, ¿producirá la fuerza 
necesaria para levantar a un automóvil de una tonelada del otro lado?

In [28]:
g = 9.81 # gravedad en m/s^2
area_pedal = 0.01 # area del pedal chico en m^2
area_prensa = 3 # area del pedal grande en m^2
p_90 = 90 # peso en kg

# calculo de la fuerza del pedal chico
def principio_de_pascal(masa, area):
    return ((masa * g) / area)

fuerza_1 = principio_de_pascal(p_90, area_pedal)
print("Fuerza del pedal chico:", fuerza_1, "N")

def regla_de_tres(a, b, c):
    return (a * b) / c 

# calculo de la fuerza producida del otro lado con la persona de 90 kg
fuerza_producida = regla_de_tres(fuerza_1, area_pedal, area_prensa)
print("Fuerza producida del otro lado del pedal con la persona de 90 kg:", fuerza_producida, "N")



# calculo para levantar un automovil de 1000 kg con una persona de 75 kg
p_75 = 75 # peso en kg

# calculo de la fuerza del pedal chico
fuerza_1 = principio_de_pascal(p_75, area_pedal)
print("Fuerza del pedal chico:", fuerza_1, "N")

# calculo de la fuerza producida del otro lado con la persona de 75 kg
fuerza_producida = regla_de_tres(fuerza_1, area_pedal, area_prensa)
print("Fuerza producida del otro lado del pedal con la persona de 75 kg:", fuerza_producida, "N")

fuerza_para_levantar_auto = 1000 * g
print("Fuerza para levantar el auto:", fuerza_para_levantar_auto, "N")

es_suficiente = fuerza_para_levantar_auto <= fuerza_producida

print(f"¿Es suficiente la fuerza producida para levantar el auto? {fuerza_producida}N >= {fuerza_para_levantar_auto}N: {es_suficiente}")



Fuerza del pedal chico: 88290.0 N
Fuerza producida del otro lado del pedal con la persona de 90 kg: 294.3 N
Fuerza del pedal chico: 73575.0 N
Fuerza producida del otro lado del pedal con la persona de 75 kg: 245.25 N
Fuerza para levantar el auto: 9810.0 N
¿Es suficiente la fuerza producida para levantar el auto? 245.25N >= 9810.0N: False


# Datos

$A_1 = 10cm = 10^{2} cm = 100cm = 0.01 m^{2}$

$A_2 = 3 m^{2}$

$g = 9.81 m/s^2$

$m_1 = 90 kg$

$m_2 = 75 kg$

# Solución

## Operaciones en base al código previo

Calculo de la fuerza producida con la persona de 90 kg:

$F_1 = \frac{m_1g}{A_1} = \frac{90(9.81)}{0.01} = 88290 N$

$F_2 = \frac{F_1 \cdot A_1}{A_2} = \frac{88290\cdot0.01}{3} = 294.3 N$


Calculo para levantar el automovil de una tonelada con la persona de 75 kg:

$F_1 = \frac{m_2g}{A_1} = \frac{75 \cdot 9.81}{0.01} = 73573.25 N$

$F_2 = \frac{F_1 \cdot A_1}{A_2} = \frac{73573.25\cdot0.01}{3} = 245.244 N$

Fuerza necesaria para levantar el automovil de una tonelada:

$F_{auto} = 1000 \cdot 9.81 = 9810 N$


Es decir, no se puede levantar el automovil con la persona de 75 kg

$\therefore F_2 < F_{auto}$


----

# Principio de Bernoulli

## Problema

Por una tubería de 3.81 cm de diámetro circula agua a una velocidad de 3 m/s. 
En una parte de la tubería hay un estrechamiento y el diámetro es de 2.54 cm.

¿Qué velocidad llevará el agua en ese punto? 

Con base al simulador dado por VASCAK 
muestra el esquema que se relaciona con este problema https://www.vascak.cz/data/android/physicsatschool/template.php?f=mech_bernoulli&l=es

In [30]:
diametro_1 = 0.0381 # diametro de la tuberia 1 en m
diametro_2 = 0.0254 # diametro de la tuberia 2 en m
velocidad_del_agua = 3 # velocidad del agua en m/s

# calculo del area de la tuberia 1
def area(diametro):
    return np.pi * ((diametro / 2) ** 2)

area_1 = area(diametro_1)
print("Area de la tuberia 1:", area_1, "m^2")

# calculo del area de la tuberia 2
area_2 = area(diametro_2)
print("Area de la tuberia 2:", area_2, "m^2")

# calculo del caudal
def caudal(a_1, a_2, v):
    return a_1 * v / a_2

caudal = caudal(area_1, area_2, velocidad_del_agua)
print("Caudal:", caudal, "m^3/s")

Area de la tuberia 1: 0.00114009182796937 m^2
Area de la tuberia 2: 0.0005067074790974977 m^2
Caudal: 6.750000000000001 m^3/s


# Datos

$D_1 = 3.81 cm = 0.0381 m$

$D_2 = 2.54 cm = 0.0254 m$

$v_1 = 3 m/s$

# Solución

## Operaciones en base al código previo

Formula de continuidad:

$A_1V_1 = A_2V_2$

$\therefore V_2 = \frac{A_1V_1}{A_2}$

Calculo de las areas:

$A_1 = \pi r^2 = \pi \frac{(D_1)^{2}}{2} = \pi\frac{(0.0381)^2}{2} = 0.00114 m^2$

$A_2 = \pi r^2 = \pi \frac{(D_2)^{2}}{2} = \pi\frac{(0.0254)^2}{2} = 0.00051 m^2$

Calculo de la velocidad:

$V_2 = \frac{A_1V_1}{A_2} = \frac{0.00114 \cdot 3}{0.00051} = 6.75 m/s$

## Simulación

![Bernoulli](principio-de-bernoulli.png)